src/amo/components/AddonCompatibilityError/index.js

```diff
@@ -53,8 +54,12 @@ export class AddonCompatibilityErrorBase extends React.Component {
 
     if (reason === INCOMPATIBLE_NOT_FIREFOX) {
       message = i18n.sprintf(i18n.gettext(`You need to
-        <a href="%(downloadUrl)s">download Firefox</a> to install this add-on.`
+        <a href="%(downloadUrl)s">download Firefox</a> to install this
+        add-on.`
       ), { downloadUrl });
+    } else if (reason === INCOMPATIBLE_LEGACY_EXTENSION) {
+      message = i18n.gettext(`This is a legacy extension that isn't
+        compatible with your version of Firefox.`);
```

Comment:
I don't think this should use the word *legacy* because version incompatibility applies to future add-ons too. Jorge had this suggestion in https://github.com/mozilla/addons-server/issues/5277#issuecomment-315185528 : "This add-on is not compatible with your version of Firefox."